File: leetcode/src/main/java/number/Base7.java
package number;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * https://leetcode.com/problems/base-7/
 *
 * Biểu diễn số n dưới dạng cơ số 7
 */
public class Base7 {

    public String convertToBase7_OK(int n) {
        return Integer.toString(n, 7);
    }

    public String convertToBase7(int n) {
        if (n == 0) return "0";
        StringBuilder result = new StringBuilder();
        int m = Math.abs(n);
        while (m > 0) {
            result.append(m % 7);
            m /= 7;
        }
        return (n < 0 ? "-" : "") + result.reverse().toString();
    }

    @Test
    public void test() {
        assertEquals("202", convertToBase7(100));
        assertEquals("0", convertToBase7(0));
        assertEquals("-10", convertToBase7(-7));
    }
}

File: py-algorithm/number/base_7.py
import unittest
import numpy as np

"""
    https://leetcode.com/problems/base-7/

    Biểu diễn số n dưới dạng cơ số 7.
"""
class Solution(unittest.TestCase):

    @staticmethod
    def convert_to_base_7(n: int) -> str:
        return np.base_repr(n, base=7)

    @staticmethod
    def convertToBase7(n: int) -> str:
        if n == 0:
            return "0"

        result = []
        m = abs(n)
        while m > 0:
            result.append(str(m % 7))
            m //= 7

        return ('-' if n < 0 else '') + ''.join(result[::-1])

    def test(self):
        self.assertEqual("0", self.convertToBase7(0))
        self.assertEqual("202", self.convertToBase7(100))
        self.assertEqual("-10", self.convertToBase7(-7))


if __name__ == '__main__':
    unittest.main()
